eclair-core/src/main/scala/fr/acinq/eclair/payment/handlers/DefaultHandler.scala

```diff
@@ -14,35 +14,36 @@
  * limitations under the License.
  */
 
-package fr.acinq.eclair.payment
+package fr.acinq.eclair.payment.handlers
 
-import akka.actor.{Actor, ActorLogging, ActorRef, PoisonPill, Props, Status}
+import akka.actor.{ActorContext, ActorRef, PoisonPill, Status}
+import akka.event.LoggingAdapter
 import fr.acinq.bitcoin.{ByteVector32, Crypto}
 import fr.acinq.eclair.channel.{CMD_FAIL_HTLC, CMD_FULFILL_HTLC, Channel}
 import fr.acinq.eclair.db.{IncomingPayment, IncomingPaymentStatus}
 import fr.acinq.eclair.payment.PaymentLifecycle.ReceivePayment
 import fr.acinq.eclair.payment.Relayer.FinalPayload
-import fr.acinq.eclair.wire._
+import fr.acinq.eclair.payment.{MultiPartPaymentHandler, PaymentReceived, PaymentRequest}
+import fr.acinq.eclair.wire.IncorrectOrUnknownPaymentDetails
 import fr.acinq.eclair.{CltvExpiry, MilliSatoshi, NodeParams, randomBytes32}
 
-import scala.concurrent.ExecutionContext
 import scala.util.{Failure, Success, Try}
 
+
 /**
  * Simple payment handler that generates payment requests and fulfills incoming htlcs.
  *
  * Created by PM on 17/06/2016.
  */
-class LocalPaymentHandler(nodeParams: NodeParams) extends Actor with ActorLogging {
+class DefaultHandler(nodeParams: NodeParams) extends ReceiveHandler {
 
-  import LocalPaymentHandler._
+  import DefaultHandler._
 
-  implicit val ec: ExecutionContext = context.system.dispatcher
-  val paymentDb = nodeParams.db.payments
+  val paymentsDb = nodeParams.db.payments
+  var pendingPayments: Map[ByteVector32, (ByteVector32, ActorRef)] = Map.empty
 
-  override def receive: Receive = main(Map.empty)
+  override def handle(implicit ctx: ActorContext, log: LoggingAdapter): PartialFunction[Any, Unit] = {
 
-  def main(pendingPayments: Map[ByteVector32, (ByteVector32, ActorRef)]): Receive = {
     case ReceivePayment(amount_opt, desc, expirySeconds_opt, extraHops, fallbackAddress_opt, paymentPreimage_opt, allowMultiPart) =>
```

Comment:
`ReceivePayment` is currently defined in `PaymentLifecycle` which doesn't really make sense...
Maybe a good opportunity to move it here in the `DefaultHandler` companion object?